shared/src/main/java/com/couchbase/lite/Replicator.java

```diff
@@ -0,0 +1,617 @@
+    }
+
+
+    /**
+     * Stops a running replicator. This method returns immediately; when the replicator actually
+     * stops, the replicator will change its status's activity level to `kCBLStopped`
+     * and the replicator change notification will be notified accordingly.
+     */
+    public void stop() {
+        if (c4repl != null)
+            c4repl.stop(); // this is async; status will change when repl actually stops
+        if (reachabilityManager != null)
+            reachabilityManager.removeNetworkReachabilityListener(this);
+    }
+
+    /**
+     * The replicator's configuration.
+     *
+     * @return
+     */
+    public ReplicatorConfiguration getConfig() {
+        return config.copy();
+    }
+
+    /**
+     * The replicator's current status: its activity level and progress. Observable.
+     *
+     * @return
+     */
+    public Status getStatus() {
+        return status.copy();
+    }
+
+    /**
+     * Set the given ReplicatorChangeListener to the this replicator.
+     *
+     * @param listener
+     */
+    public void addChangeListener(ReplicatorChangeListener listener) {
+        if (listener == null)
+            throw new IllegalArgumentException();
+        changeListeners.add(listener);
+    }
+
+    /**
+     * Remove the given ReplicatorChangeListener from the this replicator.
+     *
+     * @param listener
+     */
+    public void removeChangeListener(ReplicatorChangeListener listener) {
+        if (listener == null)
+            throw new IllegalArgumentException();
+        changeListeners.remove(listener);
+    }
+
+    @Override
+    public String toString() {
+        if (desc == null)
+            desc = description();
+        return desc;
+    }
+
+    //---------------------------------------------
+    // Implementation of NetworkReachabilityListener
+    //---------------------------------------------
+    @Override
+    public void networkReachable() {
+        if (c4repl == null) {
+            Log.e(TAG, "%s: Server may now be reachable; retrying...", this);
+            retryCount = 0;
+            retry();
+        }
+    }
+
+    @Override
+    public void networkUnreachable() {
+        Log.v(TAG, "%s: Server may NOT be reachable now.", this);
+    }
+
+    //---------------------------------------------
+    // Protected level access
+    //---------------------------------------------
+
+    // - (void) dealloc
+    @Override
+    protected void finalize() throws Throwable {
+        clearRepl();
+        if (reachabilityManager != null) {
+            reachabilityManager.removeNetworkReachabilityListener(this);
+            reachabilityManager = null;
+        }
+        super.finalize();
+    }
+
+    //---------------------------------------------
+    // Private member methods (in class only)
+    //---------------------------------------------
+
+    private void _start() {
+        // Source
+        com.couchbase.litecore.Database db = config.getDatabase().internal();
```

Comment:
`feature/replicator` branch was created from the older branch before the change.